lib/pbench/server/api/resources/query_apis/datasets_owner.py

```diff
@@ -0,0 +1,101 @@
+from logging import Logger
+from typing import Iterator
+
+from pbench.server import JSONOBJECT, OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    ApiAuthorizationType,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.api.resources.query_apis import ApiContext, ElasticBulkBase
+from pbench.server.database.models.datasets import Dataset
+
+
+class DatasetOwner(ElasticBulkBase):
+    """
+    Change the "owner" authorization of a Pbench dataset which is either public or
+    needs change in ownership, by modifying the "authorization": {"owner": owner_id}
+    subdocument of each Elasticsearch document associated with the specified dataset.
+
+    POST /api/v1/datasets/change_owner/{dataset}?owner=user
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            ApiSchema(
+                ApiMethod.POST,
```

Comment:
Should this be `PUT` instead of `POST` and also should we allow `GET` as well?

Reply:
No. Probably. But see the whole "V2" API schema I wrote earlier, and we *could* overload a `POST` on a unified URL to update owner and/or access right now.

We might also use query parameters to overload a `GET /datasets/<dataset_id>` to get (and `PUT` to update) metadata based on an `?metadata` query parameter... maybe. But, again, not here. 😆